### PR TITLE
Fuzz settings in fuzz_http3serverreq

### PR DESCRIPTION
**Pull request by amirlivneh**

### CODE REVIEW

File: fuzz/fuzz_http3serverreq.cc [hunk 1]
```diff
@@ -35,9 +45,23 @@ static int send_data(nghttp3_conn *conn) {
 extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
   FuzzedDataProvider fuzzed_data_provider(data, size);
   nghttp3_callbacks callbacks{};
-  nghttp3_settings settings;
 
-  nghttp3_settings_default(&settings);
```
Comment (tatsuhiro-t) [inline fuzz/fuzz_http3serverreq.cc:40]: This should be kept because this is how we initialize nghttp3_settings in practice.

Comment (amirlivneh) [reply fuzz/fuzz_http3serverreq.cc:40]: Makes sense. I made the change.

File: fuzz/fuzz_http3serverreq.cc [hunk 2]
```diff
@@ -75,4 +97,4 @@ extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
   nghttp3_conn_del(conn);
 
   return 0;
-}
+}
```
Comment (tatsuhiro-t) [inline fuzz/fuzz_http3serverreq.cc:100]: Looks like the line separator is missing.

File: fuzz/fuzz_http3serverreq.cc [hunk 3]
```diff
@@ -4,6 +4,16 @@
...
 
+#ifdef __cplusplus
+extern "C" {
+#endif // defined(__cplusplus)
+
+#include "nghttp3_macro.h"
+
+#ifdef __cplusplus
+}
+#endif // defined(__cplusplus)
```
Comment (tatsuhiro-t) [inline fuzz/fuzz_http3serverreq.cc:15]: This has been added in #293

File: fuzz/fuzz_http3serverreq.cc (hunk 2 above)
Comment (amirlivneh) [reply fuzz/fuzz_http3serverreq.cc]: Fixed.

File: fuzz/fuzz_http3serverreq.cc (hunk 3 above)
Comment (amirlivneh) [reply fuzz/fuzz_http3serverreq.cc]: I removed it.

**Review by tatsuhiro-t — APPROVED** [review]

**Pull request merged**